packages/x-data-grid-premium/src/hooks/features/promptControl/api.ts

```diff
@@ -0,0 +1,29 @@
+import { PromptResponse } from './types';
+
+type Result<T> = { ok: false; message: string } | { ok: true; data: T };
+
+export function processPrompt(url: string, context: string, query: string) {
+  return fetch(url, {
+    mode: 'cors',
+    method: 'post',
```

Comment:
Looks more standard?

```suggestion
    method: 'POST',
```